### PR TITLE
🎉

### PR DESCRIPTION
**Pull request by BinaryMuse**
itshappening.gif

---

For posterity: In addition to some more exciting changes ✨, this PR makes some changes to the way we run tests. In particular, if a core package has an `atomTestRunner` entry in it's `package.json`, we now perform the following operations:

1. Copy the package's `node_modules` to `node_modules.bak`
2. Run `apm install` inside the package folder to install all its devDependencies
3. Run the tests as usual
4. Remove the new `node_modules` folder and rename `node_modules.bak` to `node_modules`

This ensures that all test dependencies get installed, while also ensuring that we don't corrupt CI's `node_modules` cache.

### CODE REVIEW

**Comment by BinaryMuse** [comment]
Yessss!

@as-cii, care to take a quick 👀 over this?

File: package.json [hunk 1]
```diff
@@ -127,7 +128,7 @@
     "timecop": "0.36.0",
     "tree-view": "0.217.0-7",
     "update-package-dependencies": "0.11.0",
-    "welcome": "0.36.2",
+    "welcome": "0.36.3",
```
Comment (BinaryMuse) [inline package.json]: This just needed changes to the tests (change `atom.workspace.getPanes()` to `atom.workspace.getCenter().getPanes()`).

File: script/test [hunk 2]
```diff
@@ -59,25 +63,63 @@ function runCoreRenderProcessTests (callback) {
 // Build an array of functions, each running tests for a different bundled package
 const packageTestSuites = []
 for (let packageName in CONFIG.appMetadata.packageDependencies) {
-  const packageSpecDirPath = path.join(CONFIG.repositoryRootPath, 'node_modules', packageName, 'spec')
-  if (!fs.existsSync(packageSpecDirPath)) continue
+  if (process.env.ATOM_PACKAGES_TO_TEST) {
```
Comment (as-cii) [inline script/test]: Nice!

File: package.json (hunk 1 above)
Comment (as-cii) [inline package.json]: Makes sense. 👍

**Review by as-cii — APPROVED** [review]
Looks great! ✨ Can't wait to try this out!

**Comment by simurai** [comment]
zhu-li-did-the-thing

**Comment by as-cii** [comment]
👏 👏 👏 👏 👏 👏 👏 👏 👏 👏 👏 👏

**Comment by Ben3eeE** [comment]
:tada: :tada: :tada: :tada: :tada:

**Pull request merged**